messaginginapp/src/test/java/io/customer/messaginginapp/state/InAppMessagingMiddlewaresTest.kt

```diff
@@ -0,0 +1,472 @@
+package io.customer.messaginginapp.state
+
+import io.customer.commontest.extensions.random
+import io.customer.messaginginapp.gist.data.listeners.GistQueue
+import io.customer.messaginginapp.gist.presentation.GistListener
+import io.customer.messaginginapp.state.MessageBuilderTest.createMessage
+import io.customer.messaginginapp.testutils.core.JUnitTest
+import io.customer.messaginginapp.testutils.extension.createInAppMessage
+import io.customer.sdk.core.di.SDKComponent
+import io.customer.sdk.core.util.Logger
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.verify
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.reduxkotlin.Store
+
+/**
+ * Tests for the middleware functions in InAppMessagingMiddlewares.kt,
+ * focused on behaviors related to inline in-app messaging.
+ */
+class InAppMessagingMiddlewaresTest : JUnitTest() {
+
+    private lateinit var store: Store<InAppMessagingState>
+    private lateinit var nextFn: (Any) -> Any
+    private lateinit var mockGistQueue: GistQueue
+    private lateinit var mockGistListener: GistListener
+    private lateinit var mockLogger: Logger
+
+    @BeforeEach
+    fun setupMocks() {
+        store = mockk(relaxed = true)
+        nextFn = mockk(relaxed = true)
+        mockGistQueue = mockk(relaxed = true)
+        mockGistListener = mockk(relaxed = true)
+        mockLogger = mockk(relaxed = true)
+
+        val emptyState = InAppMessagingState()
+        every { store.state } returns emptyState
+
+        SDKComponent.overrideDependency<GistQueue>(mockGistQueue)
+        SDKComponent.overrideDependency<Logger>(mockLogger)
```

Comment:
We could use default `setup` method for this?

Reply:
updated